lab-procedure.md

```diff
@@ -1,31 +1,77 @@
-
-## Part 1: Built-in inotify beacon
-
-For this part we will use Apache web server for demonstration. Let's install and configure the service using salt.
-
-In the master config (/etc/salt/master) we see the following:
-```
+# Lab outline
+
+This lab will expose students to the following:
+- Using a built in beacon/reactor to restore a modified apache configuration file
+- Installing a (pre-made) custom beacon to monitor for high packet rates over port 80
+  - Leveraging said custom beacon to deliver a notification to system administrator
+  - Then using the custom beacon to provide firewall remediation of a potential DDOS attack
+<br>
+
+## Glossary
+- [Part 0](#part-0-check-your-salt-cluster)
+  - Check your cluster setup
+- [Part 1](#part-1-install-apache-with-salt-states)
+  - Install apache with salt states
+- [Part 1.1](#part-11-manage-apache-with-salt)
+  - Manage apache manually through salt states
+- [Part 2](#part-2-monitor-apache-with-salt-beacons)
+  - Setup your first salt beacons to watch for changes to apache files
+- [Part 2.1](#part-21-observe-the-beacon-observing)
+  - Test the salt beacon in action
+- [Part 3](#part-3-act-on-beacon-reports-with-salt-reactors)
+  - Setup your first salt reactor to respond to events
+- [Part 3.1](#part-31-react-to-the-reactor-reacting)
+  - Test the salt reactor in action
+- [Part 4](#part-4-custom-beacon-to-monitor-traffic-and-send-text-message)
+  - How to add in custom beacon code to salt
+<br><br><br>
+
+***
+## Part 0: Check your salt cluster
+1) Login to your assigned [environment]().
+If you are using windows then use these [directions]().
+```
+$ ssh -i [path to provided key] ubuntu@[your master instance IP address]
+```
+
+2) Ensure your salt master is running and has no errors.
+```
+$ systemctl status salt-master
+```
+
+3) Browse the master configuration file `/etc/salt/master`.
+The listed directories under `file_roots` is where salt will look
+for states, modules, and managed files.
+```YAML
+$ cat /etc/salt/master
 file_roots:
   base:
     - /srv/salt
 ```
-This is where salt will look for states, modules, and/or other files.
 
-Let's make the /srv/salt directory and create a state to install apache.
-
-```
-mkdir /srv/salt
-cd /srv/salt
-```
+4) Check what minion keys are accepted by your master, `$ salt-key -L`,
+then check that your master can connect to all listed minions.
+```YAML
+$ salt \* test.version
+dynamic-lab000-master:
+    2019.2.2
+dynamic-lab000-minion-red:
+    2019.2.2
+dynamic-lab000-minion-blue:
+    2019.2.2
 ```
-# /srv/salt/apache.sls
+The return will look similar to the one above. If your return looks questionable, please **request assistance**.
+<br><br><br>
+
+***
+## Part 1: Install apache with salt [states](https://docs.saltstack.com/en/latest/topics/tutorials/starting_states.html)
+For this part we will use salt to install and start Apache web server for the demonstration.
+
+1) Create a salt state to install apache in the `/srv/salt` directory.
```

Comment:
I would add here something like "`mkdir -p` will create the parent directories if they do not currently exist"